app/src/main/java/io/neurolab/communication/CommunicationHandler.java

```diff
@@ -30,6 +29,11 @@ public CommunicationHandler(Context context, UsbManager usbManager) {
         searchForArduinoDevice(context);
```

Comment:
@jddeep Would it be possible for you to make the constructor private and use a static field above pointing to an only instance (Singleton Design Pattern). Let's say if some other activity or class requires to communicate then we won't have the need to create separate instances every time which is what should happen logically for resource optimization.  @CloudyPadmal correct me if I am wrong.

Reply:
Also if possible please refactor the naming to USBSerialCommunicationManager or something like that as we would be having Bluetooth classes as well.

Reply:
Yup @suresh-gandhi I think it will be possible. Will make the amend. Thanks.